packages/botonic-cli/src/analytics/telemetry.ts

```diff
@@ -0,0 +1,83 @@
+import Analytics from 'analytics-node'
+
+import { ANALYTICS_WRITE_KEY, TRACKING_EVENTS } from '../constants'
+import { AnalyticsService } from '../interfaces'
+import { getSystemInformation } from '../util/processes'
+import { GlobalCredentialsHandler } from './credentials-handler'
+
+export class Telemetry {
+  analyticsService: AnalyticsService
+  globalCredentialsHandler: GlobalCredentialsHandler
+  isAnalyticsEnabled = process.env.BOTONIC_DISABLE_ANALYTICS !== '1'
+  events = TRACKING_EVENTS
+
+  constructor() {
+    this.analyticsService = new Analytics(ANALYTICS_WRITE_KEY, { flushAt: 1 })
+    this.globalCredentialsHandler = new GlobalCredentialsHandler()
+  }
+
+  createAnonymousIdIfNotExists(): string | undefined {
+    if (!this.globalCredentialsHandler.hasAnonymousId()) {
+      this.globalCredentialsHandler.refreshAnonymousId()
+    }
+    const credentials = this.globalCredentialsHandler.load()
+    return credentials?.analytics?.anonymous_id
+  }
+
+  getTelemetryProperties(properties: any): any {
+    return {
+      ...getSystemInformation(),
+      ...properties,
+    }
+  }
+
+  track(event: string, properties = {}): void {
+    if (this.isAnalyticsEnabled) {
+      const anonymousId = this.createAnonymousIdIfNotExists()
+      this.analyticsService.track({
+        event: event,
+        anonymousId: anonymousId,
+        properties: this.getTelemetryProperties(properties),
+      })
+    }
+  }
+
+  trackLoggedIn(properties = {}): void {
```

Comment:
My 2 cents about the names.
* Normally it makes sense to name events in the paste "creatED" since you normally want to launch them after the action has completed. 
* But here you launch the tracks before completing the action. So it would make more sense to name the functions in present (trackLogIn)
Since it's too late to change the literals sent to analytics, I would leave them and just add a comment explaining the misalignment in the code that defines the constants